src/Language/GraphQL/Draft/Parser.hs

```diff
@@ -479,3 +489,41 @@ between open close p = tok open *> p <* tok close
 -- `empty` /= `pure mempty` for `Parser`.
 optempty :: Monoid a => Parser a -> Parser a
 optempty = option mempty
+
+-- | Parses strings delimited by triple quotes. 
+-- http://spec.graphql.org/June2018/#sec-String-Value
+blockString :: Parser Text
+blockString = do
+  _ <- triplequotes <?> "opening triple quotes"
+  str <- T.lines . T.pack <$> AT.manyTill AT.anyChar triplequotes <?> "the body of a triple quoted string"
+  let !cleanLines = dropEmptyLines str
+  let smallest = foldr selectSmallest maxBound cleanLines -- counts indentation and selects the smallest one from all lines
+  let fixedLines = foldr (fixIndentation smallest) "" cleanLines -- removes the common indentation from all lines.
+  return (T.strip . toStrict . TB.toLazyText $ fixedLines)
+ where
+  -- used to remove the common indentation from each line.
+  fixIndentation :: Int -> Text -> TB.Builder -> TB.Builder 
+  fixIndentation smallest a acc =
+    let new = TB.fromText $ T.dropWhileEnd (== ' ') (T.drop smallest a)
+    in new <> "\n" <> acc
+  -- used to accumulate the smallest indentation
+  selectSmallest :: Text -> Int -> Int
+  selectSmallest t acc =
+    let new = countIndentation t
+     in if (not $ onlySpaces t) && new < acc
+       then new
+       else acc
+  -- used to count indentation in a single a line
+  countIndentation :: Text -> Int
+  countIndentation = fromMaybe 0 . T.findIndex (/= ' ')
+  triplequotes = AT.string "\"\"\""
+
+onlySpaces :: Text -> Bool
+onlySpaces = T.all (== ' ')
+
+dropEmptyLines :: [Text] -> [Text]
+dropEmptyLines =  
+  dropWhileEnd cond . dropWhile cond
+ where
+  cond l = T.null l || onlySpaces l
+  dropWhileEnd p = foldr (\x xs -> if p x && null xs then [] else x : xs) []
```

Comment:
What about tabs?

Reply:
Is there a way we could make the last line (with the `foldr`) easier to understand?